File: apple/Sources/ReactNativeWatchOS/Modifiers/AccessibilityRemoveTraits.swift
import SwiftUI

/// SwiftUI `.accessibilityRemoveTraits(_:)`. Removes one or more traits.
/// `traits` is a single name or an array combined into one
/// `AccessibilityTraits` (see `RNWAccessibilityTraitsParser`).
enum RNWAccessibilityRemoveTraitsModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("accessibilityRemoveTraits") { view, params, _ in
            let traits = RNWAccessibilityTraitsParser.parse(params["traits"])
            return AnyView(view.accessibilityRemoveTraits(traits))
        }
    }
}

File: src/modifiers/accessibilityRemoveTraits.ts
import { createModifier } from './createModifier';
import type { AccessibilityTrait } from './accessibilityAddTraits';

/// SwiftUI `.accessibilityRemoveTraits(_:)`. Removes one or more traits
/// previously inferred or added. Pass a single trait or an array; an array
/// is combined into one `AccessibilityTraits`.
export function accessibilityRemoveTraits(
  traits: AccessibilityTrait | AccessibilityTrait[]
) {
  return createModifier('accessibilityRemoveTraits', { traits });
}
